### ISSUE TITLE
E431: Format error in tags file

### ISSUE THREAD

**Issue by Cloudprogram** [universal-ctags/ctags#2118]
(
Thank you for contacting us.

If you are reporting an issue with the parsing output, please fill
the following template.  As your custom CTags configuration can
affect results, please always use `--options=NONE` as the first
option when running `ctags`.

Otherwise, delete the template and write your issue from scratch.
Examples may help developers understanding your issue better.

Use GitHub web interface and markdown notation.
Using mail results broken text rendering that makes
the developers go crazy.
)

*****

The name of the parser:

ctags
The command line you used to run ctags:
E:\program\u-boot-1.1.6>ctags --languages=asm,c,c++ --c-kinds=+lpxz --c++-kinds=
+lpxzLA --fields=+nitazS --extras=+qr --output-format=e-ctags -R E:\program\u-bo
ot-1.1.6
```
$ ctags --options=NONE ...
--languages=asm,c,c++ --c-kinds=+lpxz --c++-kinds=
+lpxzLA --fields=+nitazS --extras=+qr --output-format=e-ctags -R E:\program\u-bo
ot-1.1.6

The content of input file:
u-boot-1.1.6
```C
/* THIS IS AN EXAMPLE */
int
main(void)
{
	...
```

The tags output you are not satisfied with:

```
!_THIS_IS_AN_EXAMPLE
mainVoid	foo.c	/^main(void)$/;"	kind:function	line:2	language:C	typeref:typename:int	signature:(void)	roles:def
...
```

The tags output you expect:
no error
```
!_THIS_IS_AN_EXAMPLE
main	foo.c	/^main(void)$/;"	kind:function	line:2	language:C	typeref:typename:int	signature:(void)	roles:def
...
```

The version of ctags:
Universal Ctags 0.0.0(9b28d8ce), Copyright (C) 2015 Universal Ctags Team
Universal Ctags is derived from Exuberant Ctags.
Exuberant Ctags 5.8, Copyright (C) 1996-2009 Darren Hiebert
  Compiled: Apr 27 2019, 00:09:30
```
$ ctags --version
Universal Ctags 0.0.0(EXAMPLE), Copyright (C) 2015 Universal Ctags Team
Universal Ctags is derived from Exuberant Ctags.
Exuberant Ctags 5.8, Copyright (C) 1996-2009 Darren Hiebert
  Compiled: May 11 1018, 23:16:36
  URL: https://ctags.io/
  Optional compiled features: +wildcards, +regex, +iconv, +option-directory, +xpath, +json, +interactive, +sandbox, +yaml, +aspell
```

How do you get ctags binary:
https://github.com/universal-ctags/ctags-win32/releases
(
Building it locally, via GNU/Linux distribution, as BSD's package,
win32 binary taken from Universal-ctags/ctags-win32 project, macosx
binary taken from Universal-ctags/homebrew-universal-ctags project,
etc.
)
![QQ截图20190620101241](https://user-images.githubusercontent.com/49339934/59818468-ad91e100-9356-11e9-8b48-57cb50407179.jpg)
E:\program\u-boot-1.1.6>ctags --languages=asm,c,c++ --c-kinds=+lpxz --c++-kinds=
+lpxzLA --fields=+nitazS --extras=+qr -R E:\program\u-boot-1.1.6
If i run this command vim would work normally。
[u-boot-1.1.6.zip](https://github.com/universal-ctags/ctags/files/3308768/u-boot-1.1.6.zip)

**Comment by Cloudprogram** [comment]
E:\program\u-boot-1.1.6>ctags --languages=asm,c,c++ --c-kinds=+lpxz --c++-kinds=
+lpxzLA --fields=+nitazS --extras=+qr -R E:\program\u-boot-1.1.6
If i run this command vim would work normally。

**Comment by masatake** [comment]
I would like to see the diff output of two tags files of
```
ctags --languages=asm,c,c++ --c-kinds=+lpxz --c++-kinds=
+lpxzLA --fields=+nitazS --extras=+qr --output-format=e-ctags -R E:\program\u-bo
ot-1.1.6
```
, and
```
ctags --languages=asm,c,c++ --c-kinds=+lpxz --c++-kinds=
+lpxzLA --fields=+nitazS --extras=+qr -R E:\program\u-boot-1.1.6
```
.

**Comment by Cloudprogram** [comment]
> I would like to see the diff output of two tags files of
> 
> ```
> ctags --languages=asm,c,c++ --c-kinds=+lpxz --c++-kinds=
> +lpxzLA --fields=+nitazS --extras=+qr --output-format=e-ctags -R E:\program\u-bo
> ot-1.1.6
> ```
> 
> , and
> 
> ```
> ctags --languages=asm,c,c++ --c-kinds=+lpxz --c++-kinds=
> +lpxzLA --fields=+nitazS --extras=+qr -R E:\program\u-boot-1.1.6
> ```
> 
> .

when i use latest version gvim_8.1.1573_x64_signed.zip in https://github.com/vim/vim-win32-installer/releases,it would be ok.
The file in u-boot-1.1.6.zip(u-boot-1.1.6\tags and u-boot-1.1.6\tags (2).old)
u-boot-1.1.6\tags would cause error in vim.
But it is fine in newest vim.

**Comment by masatake** [comment]
When specifying -output-format=e-ctags, the output includes following lines:
```
!_TAG_FILE_FORMAT       2       /extended format; --format=1 will not append ;" to lines/
!_TAG_FILE_SORTED       1       /0=unsorted, 1=sorted, 2=foldcase/
!_TAG_OUTPUT_MODE       e-ctags /u-ctags or e-ctags/
!_TAG_PROGRAM_AUTHOR    Universal Ctags Team    //
!_TAG_PROGRAM_NAME      Universal Ctags /Derived from Exuberant Ctags/
!_TAG_PROGRAM_URL       https://ctags.io/       /official site/
!_TAG_PROGRAM_VERSION   0.0.0   /0fafb4e8/
)
...
```

The last ')' is a wrong entry obviously.
I will inspect this more.

**Comment by masatake** [comment]
I found two bugs. FIxing one hides the other.
I will make a pull request for fixing the first one first.

**Comment by masatake** [comment]
I merged a fix for this bug (#2120).
With the fix, the two tags file (with/without --format=e-ctags options) are the same for the input files (the source tree of u-boot).

Internally, one more fix is needed but in user-visible level, #2120 may be enough.
If my fix is not enough, please, reopen this.
